featurebyte/worker/task/base.py

```diff
@@ -99,3 +103,67 @@ async def execute(self) -> Any:
         """
         Execute the task
         """
+
+
+class BaseLockTask(BaseTask):
+    """
+    BaseLockTask is used to run a task with a lock. At most one task with the same lock
+    can be executed at the same time. The lock is released when the task is finished.
+    """
+
+    async def execute(self) -> Any:
+        """
+        Execute the task
+
+        Returns
+        -------
+        Any
+        """
+        lock = self.app_container.redis.lock(self.lock_key)
```

Comment:
Do we have to configure a lock timeout to handle unexpected scenarios where the lock is not released by abruptly terminated tasks?

Reply:
Any suggestion for the lock expired time? If we add the lock at the task level, it is a bit difficult to set the expire value. That's why i did not include it in the current implementation.

Reply:
Maybe we can set a large value like 24h as a default. The appropriate timeout could also be task dependent.